engine/apps/alerts/tasks/create_contact_points_for_datasource.py

```diff
@@ -45,57 +45,57 @@ def create_contact_points_for_datasource(alert_receive_channel_id, datasource_li
-            )
-            if response_info.get("status_code") == status.HTTP_404_NOT_FOUND:
-                grafana_alerting_sync_manager.alerting_config_with_respect_to_grafana_version(
-                    is_grafana_datasource,
-                    datasource.get("id"),
-                    datasource.get("uid"),
-                    client.get_alertmanager_status_with_config,
-                )
-                contact_point = grafana_alerting_sync_manager.create_contact_point(datasource)
-            elif response_info.get("status_code") == status.HTTP_400_BAD_REQUEST:
+        contact_point, response_info = grafana_alerting_sync_manager.create_contact_point(datasource)
+
+        if contact_point is None:
+            if response_info.get("status_code") == status.HTTP_400_BAD_REQUEST:
                 logger.warning(
-                    f"Failed to create contact point for integration {alert_receive_channel_id}, "
-                    f"datasource info: {datasource}; response: {response_info}"
+                    f"Create CP task: Failed to create contact point for integration {alert_receive_channel_id}, "
+                    f"datasource info: {datasource}; response: {response_info}. "
+                    f"Got 400 Bad Request, exclude from retry list."
                 )
                 continue
-        else:
-            contact_point = grafana_alerting_sync_manager.create_contact_point(datasource)
-        if contact_point is None:
             logger.warning(
-                f"Failed to create contact point for integration {alert_receive_channel_id} due to getting wrong "
-                f"config, datasource info: {datasource}; response: {response_info}. Retrying"
+                f"Create CP task: Failed to create contact point for integration {alert_receive_channel_id}, "
+                f"datasource info: {datasource}; response: {response_info}. Retrying"
             )
-            # Failed to create contact point due to getting wrong alerting config.
-            # Add datasource to list and retry to create contact point for it again
+            # Failed to create contact point. Add datasource to list and retry to create contact point for it again
             datasources_to_create.append(datasource)
 
     # if some contact points were not created, restart task for them
-    if datasources_to_create:
-        schedule_create_contact_points_for_datasource(alert_receive_channel_id, datasources_to_create)
+    if datasources_to_create and retry_counter < RETRY_LIMIT:
+        logger.debug(
+            f"Create CP task: Retry to create contact points for integration {alert_receive_channel_id}, "
+            f"retry counter: {retry_counter}, datasource list {len(datasource_list)}"
+        )
+        retry_counter += 1
+        schedule_create_contact_points_for_datasource(alert_receive_channel_id, datasources_to_create, retry_counter)
     else:
```

Comment:
Wondering if we should have more visibility (logs or exception) about an exhausted retries scenario?

Reply:
I improved logging here to get more information about every step and I also I changed 'restart' task to 'retry' for better visibility.